server/server_test.go

```diff
@@ -213,6 +213,50 @@ func TestGetConnectURLs(t *testing.T) {
 	}
 }
 
+func TestClusterClientAdvertiseConnectURL(t *testing.T) {
+	opts := DefaultOptions()
+	opts.Port = 4222
+	opts.Cluster.ClientAdvertiseStr = "nats.example.com"
+
+	s := New(opts)
+
```

Comment:
As an habit, I would put a `defer s.Shutdown()` here, even if I manually shutdown and recreate a server in that function...